### PR TITLE
chore: Remove client subscription for delete branch

### LINKED ISSUE #13822: [Bug]: If client cancels the subscription for delete branch, application is partially deleted
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When client cancels the subscription for delete branch application is partially deleted, this issue becomes significant when the application have lot of resources attached. As we don't have transactions in place we end up with stale entries in DB.

### Steps To Reproduce

1. Create application with more than 20 CRUD pages
2. Connect app to git with say default branch main
3. Create new branch f1 from main
4. Checkout to main
5. Delete branch f1 and immediately switch to homepage to cancel the client subscription

### Public Sample App

_No response_

### Version

Cloud: v1.6.26

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

> Delete branch can take more time if we are trying to delete the branch with lot of resources and the client may cancel the request. This leads to the flow getting stopped midway producing corrupted DB objects. The PR ensures that even though the client may have canceled the flow, the delete branch should proceed uninterrupted, and whenever the client runs sync branch flow valid branch list will be available.

Fixes #13822

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> JUnit TC
> Manual testing

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java [hunk 1]
```diff
@@ -64,6 +64,7 @@
 import org.springframework.core.io.buffer.DataBuffer;
 import org.springframework.core.io.buffer.DataBufferUtils;
 import org.springframework.core.io.buffer.DefaultDataBufferFactory;
+import org.springframework.data.mongodb.core.ReactiveMongoOperations;
 import org.springframework.http.HttpMethod;
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java:68]: Can we remove the unused import statements?

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java [hunk 2]
```diff
@@ -82,9 +83,12 @@
 import java.time.Instant;
 import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java:90]: Can we remove the unused import statements?

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java [hunk 3]
```diff
@@ -146,6 +150,9 @@ public class GitServiceTest {
     @MockBean
     PluginExecutorHelper pluginExecutorHelper;
 
+    @Autowired
+    ReactiveMongoOperations mongoOperations;
+
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java:155]: Same here as well.

**Review by AnaghHegde — COMMENTED** [review]
Minor nit, rest LGTM

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java (hunk 2 above)
Comment (abhvsn) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java:89]: These are being used in added testcase

**Review by AnaghHegde — APPROVED** [review]

**Pull request merged**